src/main/java/sirius/biz/util/ArchiveExtractor.java

```diff
@@ -0,0 +1,313 @@
+            if (isSevenZipEnabled()) {
+                supportedExtensions = Arrays.stream(ArchiveFormat.values())
+                                            .map(archiveFormat -> archiveFormat.getMethodName().toLowerCase())
+                                            .collect(Collectors.toSet());
+            } else {
+                supportedExtensions = Collections.singleton(ZIP_EXTENSION);
+            }
+        }
+
+        return Collections.unmodifiableSet(supportedExtensions);
+    }
+
+    private boolean isSevenZipEnabled() {
+        if (sevenZipEnabled == null) {
+            checkAndInitSevenZip();
+        }
+
+        return sevenZipEnabled;
+    }
+
+    @SuppressWarnings("java:S1181")
+    @Explain(
+            "We really want to catch Throwable here as we're calling native code and we want to simply suppress using 7-ZIP if any trouble occurs.")
+    private void checkAndInitSevenZip() {
+        if (Sirius.isFrameworkEnabled(FRAMEWORK_SEVEN_ZIP)) {
+            try {
+                Log.SYSTEM.INFO("Loading 7-ZIP native libraries to support all breeds of archive formats...");
+                SevenZip.initSevenZipFromPlatformJAR();
+                sevenZipEnabled = true;
+            } catch (Throwable e) {
+                Exceptions.handle()
+                          .to(Log.SYSTEM)
+                          .error(e)
+                          .withSystemErrorMessage("Failed to initialize 7-Zip: %s (%s)")
+                          .handle();
+                sevenZipEnabled = false;
+            }
+        } else {
+            sevenZipEnabled = false;
+        }
+    }
+
+    /**
+     * Checks if the file extension is an archive which can be processed by this class.
+     *
+     * @param fileExtension the extension to check
+     * @return <tt>true</tt> when this file is an archive, <tt>false</tt> otherwise
+     */
+    public boolean isArchiveFile(@Nullable String fileExtension) {
+        return fileExtension != null && getSupportedFileExtensions().contains(fileExtension.toLowerCase());
+    }
+
+    /**
+     * Determines if the given archive is a ZIP file.
+     *
+     * @param fileExtension the file extension of the file to check
+     * @return <tt>true</tt> if the given file is a ZIP file, <tt>false</tt> otherwise
+     */
+    public boolean isZIPFile(@Nullable String fileExtension) {
+        return ZIP_EXTENSION.equalsIgnoreCase(fileExtension);
+    }
+
+    /**
+     * Iterates over the items of an archive file.
+     *
+     * @param filename              the filename of the archive to extract
+     * @param archiveFile           the archive file to extract
+     * @param filter                determines which files will be processed
+     * @param extractedFileConsumer invoked for each extracted file. May return <tt>false</tt> to abort processing or
+     *                              <tt>true</tt> to continue
+     */
+    public void extract(String filename,
```

Comment:
Wouldn't it be better to pass the file extension as parameter and get the proper filename from the java.io File when needed?

Reply:
this is a temporary file and will most probably has a random filename

Reply:
I added an appropriate comment in the JavaDoc. (One could argue that most probably the file extension would match, but this approach seems a bit cleaner.